### DISCUSSION TITLE
Set ItemContainerTheme to own ControlTheme

### DISCUSSION THREAD

**Discussion by JaykeBird** [AvaloniaUI/Avalonia#14391] [Q&A] [answered]
Good morning,

To put it simply, my question is that I have a ControlTheme that I'm making for a MenuItem in XAML, and I'm trying to set that MenuItem's ItemContainerTheme to that same ControlTheme, like so:

```xaml
<ControlTheme x:Key="SsuiSubItem" TargetType="MenuItem">
    <Setter Property="ItemContainerTheme" Value="{StaticResource SsuiSubItem}" />
    <!-- rest of theme -->
</ControlTheme>
```
but I end up getting a `KeyNotFoundException` saying it can't find SsuiSubItem. I've tried a few other methods, like `{TemplateBinding Theme}`, or creating another ControlTheme that's BasedOn the actual ControlTheme and using that as the ItemContainerTheme, but none of those seem to work.

How can I accomplish what I'm looking for?

--------------------------

**More context:**

I'm creating a control library with various new controls and extended versions of existing controls. I don't want to override the default `{x:Type MenuItem}` ControlTheme, in case someone wants to use one of my other controls but still use the standard Avalonia Menu's appearance. I'd prefer not having to create my own MenuItem type, but I suppose I will have to if that's the only option left.

For WPF, I was able to accomplish this by adding a MenuItem style as a resource in my Menu control's style (Style being basically the WPF equivalent to the ControlTheme):

``` xaml
<Style TargetType="{x:Type flat:Menu}">
    <Style.Resources>
        <Style TargetType="{x:Type MenuItem}">
            <!-- custom MenuItem theme/template in here -->
        </Style>
        <Style x:Key="{x:Static MenuItem.SeparatorStyleKey}" TargetType="Separator">
            <!-- custom style/template for separators in menus -->
        </Style>
     </Style.Resources>
     <!-- menu template -->
</Style>
```

but the equivalent doesn't seem to work in Avalonia:

```xaml
<ControlTheme x:Key="{x:Type flat:Menu}" TargetType="flat:Menu">
    <ControlTheme.Resources>
        <ControlTheme x:Key="{x:Type MenuItem}" TargetType="MenuItem">
            <!-- custom MenuItem theme/template in here -->
        </ControlTheme>
    </ControlTheme.Resources>
    <!-- menu template -->
</ControlTheme>
```

I would definitely be open to this approach, if this is available.

I have found this discussion #10060 but my use case seems different from theirs. I haven't been able to set a class in a ControlTheme.

**Comment by rabbitism** [comment, marked as answer]
Bug i think this is a circular reference issue, you are using a theme key before the definition finished. May be you should try to assign as dynamic resource.

**Reply by JaykeBird** [reply]
This seems to have done the trick! Probably a pretty simple change in retrospect, but DynamicResource was not a thing that crossed my mind. Either way, thank you!
